desktop-widgets/modeldelegates.cpp

```diff
@@ -254,9 +255,16 @@ TankUseDelegate::TankUseDelegate(QObject *parent) : QStyledItemDelegate(parent)
 
 QWidget *TankUseDelegate::createEditor(QWidget *parent, const QStyleOptionViewItem &, const QModelIndex &) const
 {
+	struct divecomputer *currentDc;
+	if (DivePlannerPointsModel::instance()->currentMode() != DivePlannerPointsModel::NOTHING)
+		currentDc = &displayed_dive.dc;
+	else
+		currentDc = get_dive_dc(current_dive, dc_number);
+	bool isCcrDive = currentDc ? currentDc->divemode == CCR : true;
```

Comment:
If `currendDc` is null, this should probably exit early. The only time this can happen is if `current_dive` is null itself. In that case we don't want to display anything - do we?

So I would suggest a
```
        if (!currentDc)
               return new QComboBox(parent);
```
The line above then simplifies to
```
        bool isCcrDive = currentDc->divemode == CCR;
```